app/models/scc_repository.rb

```diff
@@ -21,13 +22,10 @@ def pretty_name
 
   def token_changed_callback
     User.current ||= User.anonymous_admin
-    scc_products.where.not(product: nil).find_each do |sp|
-      reponame = uniq_name(sp)
-      repository = sp.product.root_repositories.find_by(name: reponame)
-      unless repository.nil? || repository.url == full_url
-        ::Foreman::Logging.logger('foreman_scc_manager').info "Update URL-token for repository '#{reponame}'."
-        ForemanTasks.async_task(::Actions::Katello::Repository::Update, repository, url: full_url)
-      end
-    end
+    repo = self.katello_root_repository
+    return if repo.nil? || repo.url == full_url
```

Comment:
if its nil, this is a error? should we log it then?

Reply:
It is not an error. The callback is executed if the SccRepository is updated and this can also happen if there is no Katello Repo associated.